uitools/toolkitcpp.cmake

```diff
@@ -0,0 +1,78 @@
+# Copyright 2012-2024 Esri
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+include_directories(${CMAKE_CURRENT_LIST_DIR}/cpp
+                    ${CMAKE_CURRENT_LIST_DIR}/cpp/Esri/ArcGISRuntime/Toolkit
+                    ${CMAKE_CURRENT_LIST_DIR}/register
+                    ${CMAKE_CURRENT_LIST_DIR}/register/Esri/ArcGISRuntime/Toolkit)
+
+set(CPPPATH ${CMAKE_CURRENT_LIST_DIR}/cpp/Esri/ArcGISRuntime/Toolkit)
+
+set(REGISTERPATH ${CMAKE_CURRENT_LIST_DIR}/register/Esri/ArcGISRuntime/Toolkit)
+
+set(TOOLKITCPP_SOURCES
+    ${CPPPATH}/AuthenticationController.cpp
+    ${CPPPATH}/BasemapGalleryController.cpp
+    ${CPPPATH}/BasemapGalleryItem.cpp
+    ${CPPPATH}/BookmarksViewController.cpp
+    ${CPPPATH}/BookmarkListItem.cpp
+    ${CPPPATH}/CoordinateConversionConstants.cpp
+    ${CPPPATH}/CoordinateConversionController.cpp
+    ${CPPPATH}/CoordinateConversionOption.cpp
+    ${CPPPATH}/CoordinateConversionResult.cpp
+    ${CPPPATH}/CoordinateOptionDefaults.cpp
+    ${CPPPATH}/FloorFilterController.cpp
+    ${CPPPATH}/FloorFilterFacilityItem.cpp
+    ${CPPPATH}/FloorFilterLevelItem.cpp
+    ${CPPPATH}/FloorFilterSiteItem.cpp
+    ${CPPPATH}/LocatorSearchSource.cpp
+    ${CPPPATH}/Internal/BasemapGalleryImageProvider.cpp
+    ${CPPPATH}/Internal/GenericListModel.cpp
+    ${CPPPATH}/Internal/GenericTableProxyModel.cpp
+    ${CPPPATH}/Internal/MetaElement.cpp
```

Comment:
Is there a way in CMake to basically just add anything in that directly with a wildcard *.cpp?

Reply:
There is this: https://cmake.org/cmake/help/latest/command/file.html#glob
But here is a note from the CMake documentation on its usage

> Note: We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.

Reply:
fair enough, we shouldn't go against best practices

Reply:
In that case we should follow recommend practices and just list all files manually. Unfortunate, but we're not adding/removing lots of files to the toolkit right now.